src/shared/dlt_common.c

```diff
@@ -2529,7 +2534,7 @@ int dlt_buffer_push3(DltBuffer *buf,
         free_size = buf->size - write + read;
 
     /* check size */
-    if (free_size < (int)(sizeof(DltBufferBlockHead) + size1 + size2 + size3)) {
+    while (free_size < (int)(sizeof(DltBufferBlockHead) + size1 + size2 + size3)) {
```

Comment:
Thank you @schrewe . I think we should also update the free_size variable inside the while loop (or use do-while) to prevent the infinitive loop for this case :)